### DISCUSSION TITLE
Shadowsocks Android client failed to work

### DISCUSSION THREAD

**Discussion by strngtn** [shadowsocks/shadowsocks-android#2869] [Q&A]
**To Reproduce**
1. Go to Shadowsocks Android client
2. Create a profile, fill address/port/password/encryption and left all others as default.
3. Activate on the profile
4. Try open m.facebook.com in Firefox
5. Click "Test" button at the bottom of Shadowsocks Android.

**Expected behavior**
4. Open facebook homepage successfully.
5. Test succeeds.

**Actual behavior**
4. Fails to open facebook home page in browser
5. A toast popup which complaints "Unable to resolve host cp.cloudflare.com, no address associated with hostname"

**Screenshots**

![WechatIMG876](https://user-images.githubusercontent.com/369280/156285480-8b815f1d-59c0-424a-95ed-6d3c952df935.jpeg)
![WechatIMG877](https://user-images.githubusercontent.com/369280/156285491-f4f28c49-8d6e-4132-a952-9824b0e8b6d4.jpeg)
![WechatIMG875](https://user-images.githubusercontent.com/369280/156285496-bf562c63-c0b3-4a20-a90a-7b63086fdc22.jpeg)

**Smartphone (please complete the following information):**
 - Android/Chrome OS version:  Android 11
 - Device: Oneplus 8
 - Version: 5.2.6
 - Last version that did not exhibit the issue: 5.0.x

**Configuration**
Put an `x` inside the [ ] that applies.
* [x] IPv4 server address
* [x] IPv6 server address
* [x] Client IPv4 availability
* [x] Client IPv6 availability
* Encrypt method: chacha20-ietf-poly1305
* Route
  * [x] All

//  Not sure what this config means. 
// All DNS related settings are default values in both android client shadowsocks-rust server.
* Remote DNS:  dns.google

// Not found this settings in client side. 
* [ ] DNS over UDP

**Additional context**

1. Confirms that the private DNS in Android Phone is disabled.
2. Confirms that connection via v2ray plugin works well.

**Comment by zhonghanwen** [comment]
I also encountered this problem.

**Comment by zhonghanwen** [comment]
I try  some version apks and i found v5.2.2 shadowsocks client can work normally!!! After v5.2.2 version,  shadowsocks client can't work.
